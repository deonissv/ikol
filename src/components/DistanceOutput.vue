<script setup lang="ts">
defineProps<{
  unit: string
  distance: number
}>()
</script>

<template>
  <div class="fit row justify-between">
    <div class="title">
      <span> Odległość [{{ unit }}]</span>
    </div>
    <div class="flex a sub">{{ distance }}</div>
  </div>
</template>

<style scoped>
.title {
  font-size: medium;
}

.sub {
  font-size: medium;
  font-weight: bold;
}
</style>
