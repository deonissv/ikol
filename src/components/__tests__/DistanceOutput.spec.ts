import { describe, it, expect } from 'vitest'
import { installQuasarPlugin } from '@quasar/quasar-app-extension-testing-unit-vitest'

import { mount } from '@vue/test-utils'
import DistanceOutput from '../DistanceOutput.vue'

installQuasarPlugin()

describe.concurrent('DistanceOutput', () => {
  it('renders', () => {
    const wrapper = mount(DistanceOutput, {
      props: {
        unit: 'txt',
        distance: 1
      }
    })
    expect(wrapper.text()).toContain('txt')
    expect(wrapper.text()).toContain('1')
  })

  it('matches snapshot', () => {
    const wrapper = mount(DistanceOutput, {
      props: {
        unit: 'txt',
        distance: 1
      }
    })
    expect(wrapper.text()).toMatchSnapshot()
  })
})
