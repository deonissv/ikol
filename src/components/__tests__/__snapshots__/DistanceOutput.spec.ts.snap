// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DistanceOutput > matches snapshot 1`] = `"Odległość [txt]1"`;
